crates/sequencer/src/sequencer.rs

```diff
@@ -564,26 +429,123 @@ where
+                                );
+
+                                // Missed DA blocks means that we produce n - 1 empty blocks, 1 per missed DA block.
+                                missed_da_blocks_count = skipped_blocks - 1;
+                            }
+                        }
+
+                        if let Err(e) = self.submit_commitment(last_finalized_height, prev_l1_height).await {
+                            error!("Sequencer error: {}", e);
+                        }
+                    }
+                },
+                // If sequencer is in test mode, it will build a block every time it receives a message
+                // The RPC from which the sender can be called is only registered for test mode. This means
+                // that evey though we check the receiver here, it'll never be "ready" to be consumed unless in test mode.
+                _ = self.l2_force_block_rx.next() => {
+                    if self.config.test_mode {
+                        if let Err(e) = self.produce_l2_block(last_finalized_block.clone(), l1_fee_rate, L2BlockMode::NotEmpty).await {
+                            error!("Sequencer error: {}", e);
+                        }
+                    }
+                },
+                // If sequencer is in production mode, it will build a block every 2 seconds
+                _ = interval.tick() => {
+                    // By default, we produce a non-empty block IFF we were caught up all the way to
+                    // last_finalized_block. If there are missed DA blocks, we start producing
+                    // empty blocks at ~2 second rate, 1 L2 block per respective missed DA block
+                    // until we know we caught up with L1.
+                    let mut l2_block_mode = L2BlockMode::NotEmpty;
+                    let mut da_block = last_finalized_block.clone();
+
+                    if missed_da_blocks_count > 0 {
+                        l2_block_mode = L2BlockMode::Empty;
+                        let needed_da_block_height = prev_l1_height + (last_finalized_height - prev_l1_height - missed_da_blocks_count);
+                        da_block = self
+                            .da_service
+                            .get_block_at(needed_da_block_height)
+                            .await
+                            .map_err(|e| anyhow!(e))?;
+                    }
+
+                    let instant = Instant::now();
+                    match self.produce_l2_block(da_block, l1_fee_rate, l2_block_mode).await {
+                        Ok(_) => {
+                            // Set the next iteration's wait time to produce a block based on the
+                            // previous block's execution time.
+                            // This is mainly to make sure we account for the execution time to
+                            // achieve consistent 2-second block production.
+                            parent_block_exec_time = instant.elapsed();
```

Comment:
ticker is not reset

Reply:
The ticker is reinstantiated on the new iteration of the loop